src/uiScreens.c

```diff
@@ -215,3 +220,152 @@ void ui_displayNetworkParamsScreen(
+	TRACE_BUFFER((uint8_t *) &marginNumerator, 8);
+	TRACE_BUFFER((uint8_t *) &marginDenominator, 8);
+
+	ASSERT(marginDenominator != 0);
+	ASSERT(marginNumerator <= marginDenominator);
+	ASSERT(marginDenominator <= MARGIN_DENOMINATOR_MAX);
+
+	char marginStr[20];
+	explicit_bzero(marginStr, SIZEOF(marginStr));
+
+	{
+		// marginPercentage is a multiple of 1/100th of 1%, i.e. the fractional part of the percentage has two digits
+		// adding marginDenominator / 2 to have a rounded result
+		uint64_t marginPercentage = (10000 * marginNumerator + (marginDenominator / 2)) / marginDenominator;
+		ASSERT(marginPercentage <= 10000);
+
+		unsigned int percentage = (unsigned int) marginPercentage;
+
+		snprintf(marginStr, SIZEOF(marginStr), "%u.%u %%", percentage / 100, percentage % 100);
+		ASSERT(strlen(marginStr) < SIZEOF(marginStr) - 1);
+	}
+
+	TRACE("%s", marginStr);
+
+	ui_displayPaginatedText(
+	        "Profit margin",
+	        marginStr,
+	        callback
+	);
+}
+
+void ui_displayOwnerScreen(
+        const pool_owner_t* owner, uint8_t networkId,
+        ui_callback_fn_t callback
+)
+{
+	{
+		// assert inputs
+		ASSERT(isValidNetworkId(networkId));
+
+		switch (owner->ownerType) {
+
+		case SIGN_TX_POOL_OWNER_TYPE_KEY_HASH:
+			ASSERT(SIZEOF(owner->keyHash) == ADDRESS_KEY_HASH_LENGTH);
+			break;
+
+		case SIGN_TX_POOL_OWNER_TYPE_PATH:
+			ASSERT(bip44_isValidStakingKeyPath(&owner->path));
+			break;
+
+		default:
+			ASSERT(false);
+		}
+	}
+
+	// we display the owner as bech32-encoded reward address for his staking key
+	uint8_t rewardAddress[1 + ADDRESS_KEY_HASH_LENGTH];
+	{
+		if (owner->ownerType == SIGN_TX_POOL_OWNER_TYPE_PATH) {
+			addressParams_t rewardAddressParams = {
+				.type = REWARD,
+				.networkId = networkId,
+				.spendingKeyPath = owner->path,
+				.stakingChoice = NO_STAKING,
+			};
+
+			deriveAddress(
+			        &rewardAddressParams,
+			        rewardAddress,
+			        SIZEOF(rewardAddress)
+			);
+		} else {
+			constructRewardAddress(
+			        networkId,
+			        owner->keyHash, SIZEOF(owner->keyHash),
+			        rewardAddress, SIZEOF(rewardAddress)
+			);
+		}
+	}
+
+	char ownerDescription[MAX_PATH_LENGTH + MAX_HUMAN_ADDRESS_SIZE + 1];
+	explicit_bzero(ownerDescription, SIZEOF(ownerDescription));
+
+	if (owner->ownerType == SIGN_TX_POOL_OWNER_TYPE_PATH) {
+		bip44_printToStr(&owner->path, ownerDescription, SIZEOF(ownerDescription));
+	}
+
+	{
+		// add owner (represented as bech32-encoded reward account for owner's staking key)
+		size_t pathLength = strlen(ownerDescription);
+		ASSERT(pathLength <= MAX_PATH_LENGTH);
+		ASSERT(pathLength <= SIZEOF(ownerDescription) - 2);
+
+		if (pathLength > 0) {
+			// add a space
+			ownerDescription[pathLength++] = ' ';
+			ASSERT(pathLength <= SIZEOF(ownerDescription) - 1);
+			ownerDescription[pathLength] = '\0';
+			ASSERT(pathLength == strlen(ownerDescription));
+		}
+
+		size_t ownerAccountLength = humanReadableAddress(
```

Comment:
I think I would prefer style `strLen += humanRe...()`, that way you can perform multiple appends with

```
len += strAppend(str+len, ..., )
len += strAppend(...., )
```